tests/rptest/clients/rpk.py

```diff
@@ -120,17 +121,28 @@ def __init__(self,
         self._tls_cert = tls_cert
 
     def create_topic(self, topic, partitions=1, replicas=None, config=None):
-        cmd = ["create", topic]
-        cmd += ["--partitions", str(partitions)]
-        if replicas is not None:
-            cmd += ["--replicas", str(replicas)]
-        if config is not None:
-            cfg = [f"{k}:{v}" for k, v in config.items()]
-            for it in cfg:
-                cmd += ["--topic-config", it]
-        output = self._run_topic(cmd)
-        self._check_stdout_success(output)
-        return output
+        def create_topic():
+            try:
+                cmd = ["create", topic]
+                cmd += ["--partitions", str(partitions)]
+                if replicas is not None:
+                    cmd += ["--replicas", str(replicas)]
+                if config is not None:
+                    cfg = [f"{k}:{v}" for k, v in config.items()]
+                    for it in cfg:
+                        cmd += ["--topic-config", it]
+                output = self._run_topic(cmd)
+                self._check_stdout_success(output)
+                return (True, output)
+            except RpkException as e:
+                if "Kafka replied that the controller broker is -1" in str(e):
+                    return False
+                raise e
+
+        wait_until_result(create_topic,
```

Comment:
@rystsov did you intend to have something like `return wait_until_result` because originally `output` was returned from `create_topic` and there isn't much reason to use `wait_until_result` unless we're inspecting something after it completes which doesn't appear to be the case here.

Reply:
@rystsov